geometry/optimization/graph_of_convex_sets.h

```diff
@@ -187,33 +226,60 @@ class GraphOfConvexSets {
     std::pair<symbolic::Variable, solvers::Binding<solvers::Cost>> AddCost(
         const solvers::Binding<solvers::Cost>& binding);
 
-    /** Adds a constraint to this vertex, described by a symbolic::Formula @p f
-    containing *only* elements of x() as variables.
+    /** Adds a constraint to this vertex.
+    @param f must contain *only* elements of x() as variables.
+    @param component specifies the component of the problem to which the
+    constraint should be added. If kBoth, the constraint is added to both the
+    relaxation and the rounded problem. If kRelaxation, the constraint is added
+    only to the relaxation. If kRounding, the constraint is added only to the
+    rounded problem. Note that if convex_relaxation is false, the constraints
```

Comment:
Are we sure about this? Don't we want to add nonconvex constraints to the rounding problem? What happens if a nonconvex constraint is shipped to the MICP solver?